configuration/configmgr/configmgrlib/mod_templates/Operation.cpp

```diff
@@ -0,0 +1,77 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "Operation.hpp"
+#include "TemplateException.hpp"
+
+
+void Operation::addAttribute(attribute &newAttribute)
+{
+    m_attributes.emplace_back(newAttribute);
+}
+
+
+bool Operation::execute(EnvironmentMgr *pEnvMgr, Inputs *pInputs)
+{
+    bool rc = true;
+    size_t count, startIndex;
+
+    pInputs->setInputIndex(0);
+
+    //
+    // Determine the count of iterations to do
+    std::string countStr = pInputs->doValueSubstitution(m_count);
+    try {
+        count = std::stoul(countStr);
+    } catch (...) {
+        throw TemplateException("Non-numeric count found for count", false);
+    }
+
+
+    //
+    // Determine the starting index
+    std::string startIdxStr = pInputs->doValueSubstitution(m_startIndex);
+    try {
+        startIndex = std::stoul(startIdxStr);
+    } catch (...) {
+        throw TemplateException("Non-numeric count found for start index", false);
+    }
+
+    //
+    // Now execute the operation count times
+    for (size_t idx=0; idx < count && rc; ++idx)
+    {
+        //
+        // The basic steps are to assign the current index, then assig attribute values. Then call the
+        // required subclass implemented specific execute method.
+        pInputs->setInputIndex(startIndex + idx);
+        assignAttributeCookedValues(pInputs);
+        rc &= doExecute(pEnvMgr, pInputs);
```

Comment:
Can we change this to rc = rc && doExecute()? Using the bitwise operations here will give the expected result but is less clear.

Reply:
OK.